obspy/signal/trigger.py

```diff
@@ -406,16 +410,24 @@ def pk_baer(reltrc, samp_int, tdownmax, tupevent, thr1, thr2, preset_len,
     pfm = C.create_string_buffer(b"     ", 5)
     # be nice and adapt type if necessary
     reltrc = np.ascontiguousarray(reltrc, np.float32)
+    # Initiliaze CF array (MB)
+    c_float_p = C.POINTER(C.c_float)
+    cf_arr = np.ascontiguousarray(np.zeros(len(reltrc) - 1), np.float32)
```

Comment:
```suggestion
    cf_arr = np.zeros(len(reltrc) - 1, dtype=np.float32, order="C")
```

This should be the same result but less internal copies and thus faster.